Modules/ITS/src/ITSClusterTask.cxx

```diff
@@ -339,73 +328,88 @@ void ITSClusterTask::reset()
+    hClusterSizeLayerSummary[iLayer] = new TH1F(Form("Layer%d/AverageClusterSizeSummary", iLayer), Form("Layer%dAverageClusterSizeSummary", iLayer), 100, 0, 100);
     hClusterSizeLayerSummary[iLayer]->SetTitle(Form("Cluster size summary for Layer %d", iLayer));
     addObject(hClusterSizeLayerSummary[iLayer]);
     formatAxes(hClusterSizeLayerSummary[iLayer], "Cluster Size (pixels)", "counts", 1, 1.10);
     hClusterSizeLayerSummary[iLayer]->SetStats(0);
 
-    hGroupedClusterSizeLayerSummary[iLayer] = new TH1D(Form("Layer%d/AverageGroupedClusterSizeSummary", iLayer), Form("Layer%dAverageGroupedClusterSizeSummary", iLayer), 100, 0, 100);
+    hGroupedClusterSizeLayerSummary[iLayer] = new TH1F(Form("Layer%d/AverageGroupedClusterSizeSummary", iLayer), Form("Layer%dAverageGroupedClusterSizeSummary", iLayer), 100, 0, 100);
     hGroupedClusterSizeLayerSummary[iLayer]->SetTitle(Form("Cluster size summary for Layer %d", iLayer));
     addObject(hGroupedClusterSizeLayerSummary[iLayer]);
     formatAxes(hGroupedClusterSizeLayerSummary[iLayer], "Grouped Cluster Size (pixels)", "counts", 1, 1.10);
     hGroupedClusterSizeLayerSummary[iLayer]->SetStats(0);
 
-    hClusterTopologyLayerSummary[iLayer] = new TH1D(Form("Layer%d/ClusterTopologySummary", iLayer), Form("Layer%dClusterTopologySummary", iLayer), 300, 0, 300);
+    hClusterTopologyLayerSummary[iLayer] = new TH1F(Form("Layer%d/ClusterTopologySummary", iLayer), Form("Layer%dClusterTopologySummary", iLayer), 300, 0, 300);
     hClusterTopologyLayerSummary[iLayer]->SetTitle(Form("Cluster topology summary for Layer %d", iLayer));
     addObject(hClusterTopologyLayerSummary[iLayer]);
     formatAxes(hClusterTopologyLayerSummary[iLayer], "Cluster Topology (ID)", "counts", 1, 1.10);
     hClusterTopologyLayerSummary[iLayer]->SetStats(0);
 
-    if (iLayer < 3) {
+    hAverageClusterOccupancySummaryZPhi[iLayer] = new TH2F(Form("Layer%d/ClusterOccupancyZPhi", iLayer), Form("Cluster occupancy on Layer %d;z (cm);#phi (#circ);", iLayer), 200, -100., 100., 360, -180., 180.);
+    hAverageClusterOccupancySummaryZPhi[iLayer]->SetStats(0);
+    hAverageClusterOccupancySummaryZPhi[iLayer]->SetBit(TH1::kIsAverage);
+    hAverageClusterOccupancySummaryZPhi[iLayer]->GetYaxis()->SetLabelSize(0.02);
+    hAverageClusterOccupancySummaryZPhi[iLayer]->GetXaxis()->SetLabelSize(0.02);
+    if (mDoPublishDetailedSummary == 1) {
+      addObject(hAverageClusterOccupancySummaryZPhi[iLayer]);
+    }
+
+    hAverageClusterSizeSummaryZPhi[iLayer] = new TH2F(Form("Layer%d/ClusterSizeZPhi", iLayer), Form("Cluster size on Layer %d;z (cm);#phi (#circ);", iLayer), 200, -100., 100., 360, -180., 180.);
+    hAverageClusterSizeSummaryZPhi[iLayer]->SetStats(0);
+    hAverageClusterSizeSummaryZPhi[iLayer]->SetBit(TH1::kIsAverage);
+    hAverageClusterSizeSummaryZPhi[iLayer]->GetYaxis()->SetLabelSize(0.02);
+    hAverageClusterSizeSummaryZPhi[iLayer]->GetXaxis()->SetLabelSize(0.02);
+    if (mDoPublishDetailedSummary == 1) {
```

Comment:
here we could extend the `if` to the full definition of the histo. I mean, we can allocate memory only if we decide to use the histo. The same `if` will have to be added also in `monitorData` when filling the plots. This is more for the `async` usage, to save memory (if not needed then we will always enable these plots). Same for all other `if`. But here I would wait for @IsakovAD answer.

Reply:
Indeed it would help a lot with the memory usage; 

Just one point: we need to be accurate since we should also add an "if" case during the reset and delete of these objects as well.

Reply:
Done and now fine-binned plots are alway within if guards. I thounk though that in a separate PR we should make all the histograms `unique_ptr`, so that wa are more safe with delete operations